RecoTracker/MeasurementDet/plugins/MeasurementTrackerEventProducer.cc

```diff
@@ -105,6 +120,47 @@ MeasurementTrackerEventProducer::updatePixels( const edm::Event& event, PxMeasur
     }
   }
 
+  if (!theBadPixelFEDChannelsLabels.empty()) {
+    edm::Handle<PixelFEDChannelCollection> pixelFEDChannelCollectionHandle;
+    for (const edm::EDGetTokenT<PixelFEDChannelCollection>& tk: theBadPixelFEDChannelsLabels) {
+      if (!event.getByToken(tk, pixelFEDChannelCollectionHandle)) continue;
+      int i=0;
+      for (const auto& disabledChannels: *pixelFEDChannelCollectionHandle) {	
+	PxMeasurementDetSet::BadFEDChannelPositions positions;
+	for(const auto& ch: disabledChannels) {
+	  const sipixelobjects::PixelROC *roc_first=NULL, *roc_last=NULL;
+	  sipixelobjects::CablingPathToDetUnit path = {ch.fed, ch.link, 0};
+	  for (path.roc=1; path.roc<=(ch.roc_last-ch.roc_first)+1; path.roc++) {
+	    const sipixelobjects::PixelROC *roc = cablingMap.findItem(path);
+	    if (roc==NULL) continue;
+	    assert(roc->rawId()==disabledChannels.detId());
+	    if (roc->idInDetUnit()==ch.roc_first) roc_first=roc;
+	    if (roc->idInDetUnit()==ch.roc_last) roc_last=roc;
+	  }
+	  if (roc_first==NULL || roc_last==NULL) { 
+	    edm::LogError("PixelFEDChannelCollection")<<"Do not find either roc_first or roc_last in the cabling map."; 
+	    continue; 
+	  }
+	  const PixelGeomDetUnit * theGeomDet = dynamic_cast<const PixelGeomDetUnit*> (trackerGeom.idToDet(roc_first->rawId()));
+	  PixelTopology const * topology = &(theGeomDet->specificTopology());
+	  sipixelobjects::LocalPixel::RocRowCol local = {39, 25};   //corresponding to center of ROC row, col
+	  sipixelobjects::GlobalPixel global = roc_first->toGlobal(sipixelobjects::LocalPixel(local));
+	  LocalPoint lp1 = topology->localPosition(MeasurementPoint(global.row, global.col));
+	  global = roc_last->toGlobal(sipixelobjects::LocalPixel(local));
+	  LocalPoint lp2 = topology->localPosition(MeasurementPoint(global.row, global.col));
+	  LocalPoint ll(std::min(lp1.x(), lp2.x()), std::min(lp1.y(), lp2.y()), std::min(lp1.z(), lp2.z()));
+	  LocalPoint ur(std::max(lp1.x(), lp2.x()), std::max(lp1.y(), lp2.y()), std::max(lp1.z(), lp2.z()));	  
+	  positions.push_back(std::make_pair(ll, ur));
+	} // loop on channels
```

Comment:
I wonder if (eventually) these local reco details could be abstracted to a function living in some local reco package so that we'd have only a short `auto positions = getBadFedChannelPositions(disabledChannels, trackerGeom, cablingMap);` here? (like is with the bad ROC positions from `SiPixelQuality` https://github.com/cms-sw/cmssw/blob/7f1e2ac9542b948be1ab4dababcbf8334038d1fb/RecoTracker/MeasurementDet/plugins/MeasurementTrackerImpl.cc#L434)